posts/2020-11-09-ocp-log-forwarding.adoc

```diff
@@ -320,15 +322,42 @@ metadata:
 spec:
   managementState: "Unmanaged"
 ```
-*Note:* After you set this value to *"Unmanaged"*, any further changes to the _ClusterLogging_ or _LogForwarding_ instances are not automatically detected. To automatically detect changes, you must change the *managementState* field back to *"Managed"*.
 
-Next, set the *MERGE_JSON_LOG* environment variable to *true*.
+Alternatively, you can modify this file on the web console. Ensure you are in the `openshift-logging` project namespace and go to `Administration > Custom Resource Definitions > ClusterLogging > Instances > Instance > YAML`. Change the `managementState` field from "Managed" to "Unmanaged".
+
+Next, we'll need to change the  *merge_json_log* attribute of the `kubernetes.var.log.containers.**` filter to *true* in the Fluentd's config map. If you are running on Open Shift 4.3 to 4.5 this step will be different. See below for the appropriate steps.
 
+```
+[root@ocp ~]# oc edit configmap fluentd
+
+apiVersion: v1
+data:
+...
+  <filter kubernetes.var.log.containers.**>
+    @type parse_json_field
+    merge_json_log 'true'
+    preserve_json_log 'true'
+    json_fields 'log,MESSAGE'
+  </filter>
+...
+```
+
+Alternatively, you can modify this file on the web console. Ensure you are in the `openshift-logging` project namespace and go to `Workloads > Config Maps > fluentd > YAML`.  Change the value for the `merge_json_log` attribute of the `kubernetes.var.log.containers.**` filter to true.
+
+**OpenShift 4.3 to 4.5 only:** Instead of manually changing the config map you can run:
 [source]
 ----
 [root@ocp ~]# oc set env ds/fluentd MERGE_JSON_LOG=true
 ----
 
+Finally we need to force restart the Fluentd pods to pick up this new change:
+[source]
+----
+oc delete pod --selector logging-infra=fluentd
+----
+
+*Note:* After you set the `managementState` to "Unmanaged" in the _ClusterLogging_ instance, any further changes to the _ClusterLogging_ or _ClusterLogForwarder_ instances are not automatically detected. If you need to fix or modify the configuration for either you will need to change the `managementState` field back to "Managed". However, once the `managementState` is changed to "Managed" the __ClusterLogging__ instance will revert the Fluentd's config map back to the original settings. You will need to complete the above steps again to enable Fluentd to parse JSON.
```

Comment:
Suggested edit:

*Note:* After you set the `managementState` to "Unmanaged" in the _ClusterLogging_ instance, any further changes to the _ClusterLogging_ or _ClusterLogForwarder_ instances are not automatically detected. If you need to modify the configuration for either instance, you must change the `managementState` field back to "Managed". However, after the `managementState` is changed to "Managed", the __ClusterLogging__ instance  reverts the Fluentd config map back to its original settings. You must complete the previous steps again to re-enable Fluentd to parse JSON.

Reply:
done